pkg/build/webhook/generic/generic.go

```diff
@@ -44,7 +45,10 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		return nil, true, nil
 	}
 
-	if req.Body != nil && req.Header.Get("Content-Type") == "application/json" {
+	contentType := req.Header.Get("Content-Type")
+	mediaType, _, err := mime.ParseMediaType(contentType)
```

Comment:
If not supplied it will result in an error. So I'll update it to return an error only if Content-Type header is specified & unparseable, sound OK?